p2p/exchange.go

```diff
@@ -50,7 +49,7 @@ type Exchange[H header.Header[H]] struct {
 func NewExchange[H header.Header[H]](
 	host host.Host,
 	peers peer.IDSlice,
-	gater *conngater.BasicConnectionGater,
+	blacklistPeer func(id peer.ID) error,
```

Comment:
There is an interface https://github.com/libp2p/go-libp2p/blob/master/core/connmgr/gater.go

Reply:
It is different one, used as dialer interceptor

Reply:
It does not have `BlockPeer()`

Reply:
I see, that's confusing